src/library/library.cpp

```diff
@@ -188,8 +188,7 @@ Library::Library(
     // sandboxing then we will need them to give us permission.
     qDebug() << "Checking for access to user's library directories:";
     foreach (QString directoryPath, getDirs()) {
-        QFileInfo directory(directoryPath);
-        bool hasAccess = Sandbox::askForAccess(directory.canonicalFilePath());
+        bool hasAccess = Sandbox::askForAccess(directoryPath);
```

Comment:
I guess the canonical path might be used for a reason. Why not skip the ckeck if the canonical path is empty? Asking to access a nonexistent path doesn't make any sense.

Reply:
I have checked all places where it is used and any path is valid. In addition the askForAccess() is called with any path from different places inside Mixxx. The token itself is created by the QfileInfos, so It is reasonable to use the path in dialogs the user has entered in the library directory picker.